SwiftCompilerSources/Sources/Basic/Utils.swift

```diff
@@ -126,10 +126,13 @@ extension String {
   }
 
   /// Underscored to avoid name collision with the std overlay.
-  /// To be replaced with an overlay call once the CI uses SDKs built with Swift 5.8.
+  /// To be replaced with an overlay call once the CI uses Swift 5.9.
   public init(_cxxString s: std.string) {
-    self.init(cString: s.__c_strUnsafe())
-    withExtendedLifetime(s) {}
```

Comment:
Is it safe to `withExtendedLifetime`? (even when building with a 5.8 compiler)

Reply:
Yes, it is safe as long as we use the initializer that is implemented in the overlay. The overlay handles the lifetime of the `std.string` parameter.